Library/src/Web/EmscriptenWebClient/EmscriptenWebClient.cpp

```diff
@@ -70,10 +70,26 @@ void OnFetchSuccessOrError(emscripten_fetch_t* Fetch)
 
 	for (int i = 0; i < Keys->Size(); ++i)
 	{
-		csp::common::String Key = Keys->operator[](i);
-		csp::common::String Val = Headers[Key];
-
-		Payload.AddHeader(Key, Val);
+		auto Key	  = Keys->operator[](i);
+		auto Val	  = std::string(Headers[Key].c_str());
+		auto KeyValue = std::string(Key.c_str());
+		// Make Key and Val lower-case
+		std::transform(KeyValue.begin(),
```

Comment:
It seems wasteful to convert to std::string just to make a string lowercase. This can be done directly on a C-string quite easily:

```c++
auto Buffer = new char[Key.Length()];
memcpy(Buffer, Key.c_str(), Key.Length());

for (auto* Char = Buffer; *Char; Char++)
{
    *Char = tolower(*Char);
}

Key = String(Buffer);
```

Better yet, you could add a `MakeLower` function to `String` that calls a function in `String::Impl` that directly modifies the internal c-string so you don't need to make a mutable copy of it.

Reply:
for the sake of getting this over the line, i'll leave this for a tech debt ticket

Reply:
I would rather we not introduce tech debt and fix this before getting this merged.

Reply:
sorted

Reply:
ended up taking out due to artifacts being present in the output string causing verification to fail.

Reply:
Ok, its not worth delaying the fix for any further. Thanks for taking a look though!

Reply:
I've tested the latest personal build and it works, can we resolve these?